observability/metrics/src/main/scala/com/daml/metrics/InstrumentedExecutorServiceMetrics.scala

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics
+
+import java.util
+import java.util.concurrent.{Callable, ExecutorService, Future, TimeUnit}
+
+import com.daml.metrics.api.MetricDoc.MetricQualification.Debug
+import com.daml.metrics.api.MetricHandle.{Counter, LabeledMetricsFactory, Meter, Timer}
+import com.daml.metrics.api.{MetricDoc, MetricsContext}
+
+import scala.jdk.CollectionConverters.{CollectionHasAsScala, SeqHasAsJava}
+
+class InstrumentedExecutorServiceMetrics(factory: LabeledMetricsFactory) {
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks submitted to an instrumented executor.",
+    description = "Number of tasks that were submitted to the executor.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service: `fork_join` or `thread_pool`.",
+    ),
+  )
+  val submitted: Meter =
+    factory.meter(
+      InstrumentedExecutorServiceMetrics.Prefix :+ "submitted",
+      "Number of tasks that were submitted to the executor",
+    )
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks running in an instrumented executor.",
+    description = "The number of currently running tasks.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val running: Counter = factory.counter(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "running",
+    "The number of currently running tasks.",
+  )
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks completed in an instrumented executor.",
+    description = "The number of tasks completed by this executor",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val completed: Meter = factory.meter(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "completed",
+    "The number of tasks completed by this executor",
+  )
+
+  @MetricDoc.Tag(
+    summary = "The time that a task is idle in an instrumented executor.",
+    description =
+      "A task is considered idle if it was submitted to the executor but it has not started execution yet.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val idle: Timer = factory.timer(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "idle",
+    "The time that a task is idle in an instrumented executor. A task is considered idle if it was submitted to the executor but it has not started execution yet.",
+  )
+
+  @MetricDoc.Tag(
+    summary = "The duration of a task is running in an instrumented executor.",
```

Comment:
Nit:
```suggestion
    summary = "The duration of a task running in an instrumented executor.",
```

Reply:
@miklos-da Wouldn't it be better to remove the `of`?

Reply:
Great question :) Here are 2 hopefully better suggestions:
```suggestion
    summary = "The total time a task runs in an instrumented executor.",
```
or
```suggestion
    summary = "The duration a task runs in an instrumented executor.",
```